server/encode/encode.go

```diff
@@ -94,9 +98,33 @@ func ffmpegCommand(filePath string, profile Profile) (*exec.Cmd, error) {
 	}
 	args = append(args, profile.ffmpegOptions...)
 	if profile.forceRG {
+		aBaselineGain := 6
+		if profile.hiGainRG {
+			// This baseline gain results in final track being +3~5dB louder
+			// than Foobar2000's default ReplayGain target volume.
+			// This makes it easier to listen to music in a car, where all other
+			// sources are usually ten thousand times louder than RG-adjusted music.
+			// -- @spijet
+			aBaselineGain = 15
+		}
+		aFilters := []string{
+			fmt.Sprintf("volume=replaygain=track:replaygain_preamp=%ddB:replaygain_noclip=0", aBaselineGain),
+			"alimiter=level=disabled",
+			"asidedata=mode=delete:type=REPLAYGAIN",
+		}
+
+		// opus always forces output to 48kHz sampling rate, but we can still use upsampling
+		// to increase RG and alimiter's peak limiting precision, which is desirable in some
+		// cases. ffmpeg's `soxr` resampler is quite fast on x86-64: it takes around 5 seconds
+		// on my Ryzen 3600 to transcode an 8-minute FLAC with 2x upsample and RG applied.
+		// -- @spijet
+		if profile.upsample {
+			aFilters = append([]string{"aresample=96000:resampler=soxr"}, aFilters...)
+		}
```

Comment:
nit, if the order doesn't matter here this can be a bit simpler
```suggestion
		if profile.upsample {
			aFilters = append(aFilters, "aresample=96000:resampler=soxr")
		}
```

Reply:
The order is important here (at least from what I understand from `ffmpeg`'s audio filter chain docs), as `ffmpeg` needs to resample the audio to higher rate *before* applying RG and limiting the output. It'll downsample the audio to 48kHz later (that's what Opus encoder does unconditionally, as it's part of its standard).

Reply:
Gotcha thanks 👍